src/main/kotlin/no/skatteetaten/aurora/gillis/service/RenewService.kt

```diff
@@ -1,36 +1,39 @@
 package no.skatteetaten.aurora.gillis.service
 
-import no.skatteetaten.aurora.gillis.controller.SourceSystemException
-import no.skatteetaten.aurora.gillis.controller.blockNonNullAndHandleError
-import no.skatteetaten.aurora.gillis.service.CrawlService.RenewableCertificate
 import org.springframework.stereotype.Service
-import org.springframework.web.reactive.function.BodyInserters
 import org.springframework.web.reactive.function.client.WebClient
+import mu.KotlinLogging
+import no.skatteetaten.aurora.gillis.controller.SourceSystemException
+import no.skatteetaten.aurora.gillis.controller.handleError
+import no.skatteetaten.aurora.gillis.service.CrawlService.RenewableCertificate
+import reactor.core.publisher.Mono
 
 data class Response(
     val success: Boolean,
     val message: String
 )
 
+private val logger = KotlinLogging.logger { }
+
 @Service
 class RenewService(val client: WebClient) {
-
-    fun renew(it: RenewableCertificate): Response {
-
-        val res = client.post()
+    fun renew(renewableCertificate: RenewableCertificate): Mono<Response> {
+        return client.post()
             .uri("/v1/sts")
-            .body(BodyInserters.fromObject(it.payload))
+            .bodyValue(renewableCertificate.payload)
             .retrieve()
             .bodyToMono(Response::class.java)
-            .blockNonNullAndHandleError(sourceSystem = "boober")
-
-        if (!res.success) {
-            throw SourceSystemException(
-                message = res.message,
-                code = "200",
-                sourceSystem = "boober"
-            )
-        }
-        return res
+            .map {
+                if (!it.success) throw SourceSystemException(
+                    message = it.message,
+                    code = "200",
```

Comment:
Skal vi ha 200 her? Eller bør vi ha 500?

Reply:
Blir dette håndtert av .handleError når du kaster?

Reply:
I dette caset så er status koden fra boober 200, med success=false og message.
Muligens derfor det var satt til 200 her. Men det blir vel kanskje mer riktig å ha 500 her, evt også senere endre status kode i boober.

Ja det blir kanskje ikke helt gjenspeilet i testen `expect error when response has success false` når man ser hva som assertes på, men når man kjører den så logges det `Could not renew cert message=Failed renewing certificate statusCode=200`

Reply:
Hmm, jeg er litt usikker hva jeg foretrekker. Code kan ha to meninger her, hvis man setter den til 200 her så sier man indirekte at det gikk fint å kalle boober, men siden man fikk en Error log så sier det at boober ikke klarte det.

Så det er ikke rett frem hva som er korrekt her. Så hvis man tenker seg at code betyr om kallet til boober gikk fint og ikke selve resultatet av det. Så er det egentlig helt fint at code har verdi 200

Reply:
Jeg tror typen `SourceSystemException` er tatt fra gobo. Tanken med code i gobo var å vise response koden til kilde systemet. Ble stort sett brukt for å kunne gi noen gode log-statements i gobo.